lxd/device/gpu_physical.go

```diff
@@ -371,11 +621,20 @@ func (d *gpuPhysical) postStop() error {
 	v := d.volatileGet()
 
 	if d.inst.Type() == instancetype.Container {
-		// Remove host files for this device.
-		err := unixDeviceDeleteFiles(d.state, d.inst.DevicesPath(), "unix", d.name, "")
-		if err != nil {
-			return fmt.Errorf("Failed to delete files for device '%s': %w", d.name, err)
+		cdiID, err := cdi.ToCDI(d.config["id"])
+		if err == nil && cdiID.Empty() {
```

Comment:
Please can you make this consistent with the logic structure in startContainer and do something like:

```go
cdiID, err := cdi.ToCDI(d.config["id"])
if err != nil {
    return nil, err
}


if !cdiID.Empty() {
     err = unixDeviceDeleteFiles(d.state, d.inst.DevicesPath(), "cdi.unix", d.name, "")
     ...
}
else {
    err = unixDeviceDeleteFiles(d.state, d.inst.DevicesPath(), "unix", d.name, "")
    ...
}
```

This makes it easier to read than switching pattern in each function.

Reply:
fixed